romancal/flatfield/flat_field_step.py

```diff
@@ -20,7 +21,10 @@ def process(self, step_input):
         input_model = rdm.open(step_input)
         # Get reference file paths
         reference_file_names = {}
-        reffile = self.get_reference_file(input_model, "flat")
+        try:
+            reffile = self.get_reference_file(input_model, "flat")
+        except CrdsLookupError:
+            reffile = None
```

Comment:
If this is to cover spectral data, I think we have two other options
- Change the flat rmap in CRDS to include `exposure.type` as a selector and return N/A for spectral types
- Look at exposure.type in the beginning of the step and if it's `grism` or `prism` skip the step.
I think `CrdsLookupError` is reserved for when a value is expected from CRDS but not returned, for example because of a wrong rmap.
I would go with the second option now but we should consider the first one too.